core/src/kura.rs

```diff
@@ -350,8 +353,8 @@ impl<IO: DiskIO> BlockStore<IO> {
     /// - Filesystem access failure (HW or permissions)
     ///
     pub async fn get_block_path(&self, block_height: NonZeroU64) -> Result<PathBuf> {
-        let filename = self.get_block_filename(block_height).await?;
-        Ok(self.path.join(filename))
+        let file_string = self.get_block_filename(block_height).await?;
+        Ok(self.filename_to_path(&file_string))
     }
```

Comment:
It's a little confusing for me that `get_block_filename()` does not return valid path and needs further processing... Maybe rename it to something else or refactor somehow?